packages/expect/src/matchers.js

```diff
@@ -546,22 +532,36 @@ const matchers: MatchersObject = {
           `Not to have a nested property:\n` +
           `  ${printExpected(keyPath)}\n` +
           (valuePassed ? `With a value of:\n  ${printExpected(value)}\n` : '')
-      : () =>
-          matcherHint('.toHaveProperty', 'object', 'path', {
-            secondArgument: valuePassed ? 'value' : null,
-          }) +
-          '\n\n' +
-          `Expected the object:\n` +
-          `  ${printReceived(object)}\n` +
-          `To have a nested property:\n` +
-          `  ${printExpected(keyPath)}\n` +
-          (valuePassed ? `With a value of:\n  ${printExpected(value)}\n` : '') +
-          (traversedPath
-            ? `Received:\n  ${RECEIVED_COLOR(
-                'object',
-              )}.${traversedPath}: ${printReceived(lastTraversedObject)}`
-            : '') +
-          (diffString ? `\nDifference:\n\n${diffString}` : '');
+      : () => {
+          const diffString =
+            valuePassed && hasEndProp
+              ? diff(value, result.value, {
+                  expand: this.expand,
+                })
+              : '';
+          return (
+            matcherHint('.toHaveProperty', 'object', 'path', {
+              secondArgument: valuePassed ? 'value' : null,
+            }) +
+            '\n\n' +
+            `Expected the object:\n` +
+            `  ${printReceived(object)}\n` +
+            `To have a nested property:\n` +
+            `  ${printExpected(keyPath)}\n` +
+            (valuePassed
+              ? `With a value of:\n  ${printExpected(value)}\n`
+              : '') +
+            (hasEndProp
+              ? `Received:\n` +
+                `  ${printReceived(result.value)}` +
+                (diffString ? `\n\nDifference:\n\n${diffString}` : '')
```

Comment:
>\n\nDifference

is this double newline intended? Old code has only one.

Reply:
Ok, I see from snapshot that it actually looks better now 👍

Reply:
Yes, double newline is following pattern of `toEqual` matcher.